lib/ace/transport_app.rb

```diff
@@ -93,6 +97,10 @@ def validate_schema(schema, body)
       error = validate_schema(@schemas["execute_catalog"], body)
       return [400, error.to_json] unless error.nil?
 
+      ForkUtil.isolate do
+        @plugins.sync(body['compiler']['environment'])
```

Comment:
As a matter of principle, runtime considerations like locking and isolation should be baked into the method with the special needs.

If you need a non-isolated version of `sync` internally to the `PluginCache`, instead do this:

```
def sync(*args)
  ForkUtil.isolate { sync_core(*args) }
end

private
def sync_core(...)
  # the real implementation
end
```